controllers/argocd/configmap.go

```diff
@@ -331,9 +362,42 @@ func (r *ReconcileArgoCD) reconcileArgoConfigMap(cr *argoprojv1a1.ArgoCD) error
 	}
 
 	cm.Data[common.ArgoCDKeyOIDCConfig] = getOIDCConfig(cr)
+
+	// new format with subkeys for ResourceCustomizations
+	if c := getNewResourceCustomizations(cr); c != nil {
+		for k, v := range c {
+			cm.Data[k] = v
+		}
+	}
+
+	// old format of ResourceCustomizations, use should not be encouraged but is supported for now
 	if c := getResourceCustomizations(cr); c != "" {
+		// Emit event providing users with deprecation notice for ResourceCustomization
+		// if !ResourceCustomizationDeprecationWarningEmitted {
+		// err := argoutil.CreateEvent(r.Client, "Warning", "Deprecated", "ResourceCustomizations is deprecated, please use the new format, ResourceCustomizationsNew, instead.", "DeprecationNotice", cr.ObjectMeta, cr.TypeMeta)
+		// ResourceCustomizationDeprecationWarningEmitted = true
+		// if err != nil {
+		// 	return err
+		// }
+		// }
+
+		// Emit event providing users with deprecation notice for ResourceCustomization if not emitted already
```

Comment:
we can probably get rid of the commented code

Reply:
Yes, Please